src/coprocessor/dag/executor/scanner.rs

```diff
@@ -75,15 +82,16 @@ impl Scanner {
         self.no_more = false;
         match self.scan_mode {
             ScanMode::Backward => self.seek_key = self.range.get_end().to_vec(),
-            ScanMode::Forward => {
-                // For forward scan, we need to re-initialize the scanner.
-                self.seek_key = self.range.get_start().to_vec();
-                self.statistics_cache.add(self.scanner.get_statistics());
-                let upper_bound = Some(Key::from_raw(&self.range.end).encoded().to_vec());
-                self.scanner = store.scanner(self.scan_mode, self.key_only, upper_bound)?;
-            }
+            ScanMode::Forward => self.seek_key = self.range.get_start().to_vec(),
             _ => unreachable!(),
         };
+
+        self.statistics_cache.add(self.scanner.get_statistics());
+        let lower_bound = Some(Key::from_raw(&self.range.start).encoded().to_vec());
+        let upper_bound = Some(Key::from_raw(&self.range.end).encoded().to_vec());
```

Comment:
What happen if it reverse seek `upper_bound`?

Reply:
Inside the RocksDB, `SeekForPrev(upper_bound)` will return the largest valid key, it is equal to `SeekToLast()`. I see we use `SeekForPrev` to implement reverse seek, so the behavior is the same.